ftdc/custom_format.go

```diff
@@ -131,18 +188,50 @@ func writeDatum(time int64, prev, curr []float32, output io.Writer) {
 // The function right now does not recursively walk data structures. We assume for now that the
 // caller will only feed "already flat" structures into FTDC. Later commits will better validate
 // input and remove limitations.
-func getFieldsForItem(item any) []string {
-	var fields []string
-	rType := reflect.TypeOf(item)
-	if val := reflect.ValueOf(item); val.Kind() == reflect.Pointer {
-		rType = val.Elem().Type()
+func getFieldsForStruct(item reflect.Type) ([]string, error) {
+	flattenPtr := func(inp reflect.Type) reflect.Type {
+		for inp.Kind() == reflect.Pointer {
+			inp = inp.Elem()
+		}
+		return inp
```

Comment:
we have defined this same lambda twice. Do we want to just make it a private function?

Reply:
We have not defined the same lambda twice!

This actually tricked me as well.

Reply:
ah wow.... yeah, I missed that. 

I see it now.

Reply:
FWIW, I think one can factor the code out with a:
```
type Kinder[T any] interface {
  Kind() T
}
```

Maybe. Type inference is not really one of Go's strongsuits. And not to mention, I'd be saving 5 lines of implementation, but having to spend 3 back to create the interface.

Reply:
Agree on the two lambdas being better than any generic code. I was about to make the same "these are the same lambda" comment, too 😆 .